### ISSUE TITLE
Images never finish uploading

### ISSUE THREAD

**Issue by ericlathrop** [tootsuite/mastodon#15615]
### Expected behaviour

Attaching an image to a toot should complete and show a thumbnail in the compose box.

### Actual behaviour

The "uploading..." progress bar appears, and goes to 100%, but never completes the upload.

In the Firefox dev tools I can see a `POST /api/v2/media` request for the upload that succeeds with a `202 Accepted` response.
Then, it looks like something is polling `/api/v1/media/105603024580754479` forever, which returns a `206 Partial Content` response.
I looked at the filesystem, and found these files under `/public/system/media_attachments/files/105/603/024/580/754/479`:

```
├── original
│   └── 6cd43b37e013c9eb.jpeg
└── small
    └── 6cd43b37e013c9eb.jpeg
```

The "original" size appears to be the complete file I uploaded.

The "web" container''s docker logs look like this:

```
web_1        | [befd608a-8284-40c3-8cd2-a58321b274c4] [paperclip] Trying to link /tmp/RackMultipart20210123-12-19d22v1.JPG to /tmp/f33439f04bae2ee052089f28e7950cff20210123-12-898m6c.JPG
web_1        | [befd608a-8284-40c3-8cd2-a58321b274c4] [paperclip] Trying to link /tmp/f33439f04bae2ee052089f28e7950cff20210123-12-898m6c.JPG to /tmp/8366d2e8260a2f82acfa1babf72b894820210123-12-co6zxn.jpeg
web_1        | [befd608a-8284-40c3-8cd2-a58321b274c4] Command :: file -b --mime '/tmp/8366d2e8260a2f82acfa1babf72b894820210123-12-co6zxn.jpeg'
web_1        | [befd608a-8284-40c3-8cd2-a58321b274c4] Command :: identify -format %m '/tmp/f33439f04bae2ee052089f28e7950cff20210123-12-898m6c.JPG[0]'
web_1        | [befd608a-8284-40c3-8cd2-a58321b274c4] Command :: convert '/tmp/f33439f04bae2ee052089f28e7950cff20210123-12-898m6c.JPG[0]' -auto-orient -resize "533x300>" -quality 90 -strip +set modify-date +set create-date '/tmp/017feb72dfdb3673ae70bc208ab4781720210123-12-1l7d7fw'
web_1        | [befd608a-8284-40c3-8cd2-a58321b274c4] Command :: convert '/tmp/017feb72dfdb3673ae70bc208ab4781720210123-12-1l7d7fw' RGB:-
web_1        | [befd608a-8284-40c3-8cd2-a58321b274c4] [paperclip] Trying to link /tmp/017feb72dfdb3673ae70bc208ab4781720210123-12-1l7d7fw to /tmp/091e6f10332068dec5529098882ddc0220210123-12-981zn2
web_1        | [befd608a-8284-40c3-8cd2-a58321b274c4] [paperclip] Trying to link /tmp/f33439f04bae2ee052089f28e7950cff20210123-12-898m6c.JPG to /tmp/8366d2e8260a2f82acfa1babf72b894820210123-12-uxc0wc.jpeg
web_1        | [befd608a-8284-40c3-8cd2-a58321b274c4] Command :: file -b --mime '/tmp/8366d2e8260a2f82acfa1babf72b894820210123-12-uxc0wc.jpeg'
web_1        | [befd608a-8284-40c3-8cd2-a58321b274c4] method=POST path=/api/v2/media format=html controller=Api::V2::MediaController action=create status=202 duration=194.08 view=0.90 db=14.16
web_1        | [b18be5d9-5cd4-418e-996e-794caa54f48b] method=GET path=/api/v1/media/105603105691353135 format=html controller=Api::V1::MediaController action=show status=206 duration=4.07 view=0.94 db=0.79
web_1        | [c048dc6b-dec7-4acb-b481-5959bf7500f8] method=GET path=/api/v1/media/105603105691353135 format=html controller=Api::V1::MediaController action=show status=206 duration=3.82 view=0.89 db=0.81
web_1        | [e1d1e244-a19c-423d-a163-2c348e728871] method=GET path=/api/v1/media/105603105691353135 format=html controller=Api::V1::MediaController action=show status=206 duration=3.48 view=0.89 db=0.61
```

and the "sidekiq" container logs looks like:

```


sidekiq_1    | [paperclip] Trying to link /tmp/da146330941be3ad691f476032d63f6920210123-7-1t5gx8q.png to /tmp/019f4705c8993c97ac651c2474faba4320210123-7-1hynizj.png
sidekiq_1    | Command :: file -b --mime '/tmp/019f4705c8993c97ac651c2474faba4320210123-7-1hynizj.png'
sidekiq_1    | [paperclip] Trying to link /tmp/da146330941be3ad691f476032d63f6920210123-7-1t5gx8q.png to /tmp/885eb80ff72ebdbd1c8de59b1f9ed08120210123-7-xykpjx.png
sidekiq_1    | Command :: convert '/tmp/885eb80ff72ebdbd1c8de59b1f9ed08120210123-7-xykpjx.png' RGB:-
sidekiq_1    | [paperclip] Trying to link /tmp/885eb80ff72ebdbd1c8de59b1f9ed08120210123-7-xykpjx.png to /tmp/e290d6d4b367b2361c0cd87281317fba20210123-7-1ioe3yt.png
sidekiq_1    | [paperclip] Trying to link /tmp/885eb80ff72ebdbd1c8de59b1f9ed08120210123-7-xykpjx.png to /tmp/019f4705c8993c97ac651c2474faba4320210123-7-133o88.png
sidekiq_1    | Command :: file -b --mime '/tmp/019f4705c8993c97ac651c2474faba4320210123-7-133o88.png'
sidekiq_1    | 2021-01-23T04:23:15.674Z pid=7 tid=q0x3 class=ActivityPub::ProcessingWorker jid=e07573c3bee93f79e0fa54ac elapsed=9.315 INFO: done
sidekiq_1    | 2021-01-23T04:23:15.674Z pid=7 tid=q0x3 class=ActivityPub::ProcessingWorker jid=38f19cb1b134a4205ee04154 INFO: start
sidekiq_1    | 2021-01-23T04:23:15.718Z pid=7 tid=q0x3 class=ActivityPub::ProcessingWorker jid=38f19cb1b134a4205ee04154 elapsed=0.044 INFO: done
sidekiq_1    | 2021-01-23T04:23:15.718Z pid=7 tid=q0x3 class=ActivityPub::ProcessingWorker jid=5ad8dcf5df65375f9f133757 INFO: start
sidekiq_1    | 2021-01-23T04:23:16.222Z pid=7 tid=q0l7 class=ActivityPub::ProcessingWorker jid=fefe6b40a6789792e9815a18 elapsed=10.037 INFO: fail
sidekiq_1    | 2021-01-23T04:23:16.222Z pid=7 tid=q0l7 WARN: {"context":"Job raised exception","job":{"retry":8,"queue":"default","backtrace":true,"class":"ActivityPub::ProcessingWorker","args":[4235,"{\"@context\": \"https://www.w3.org/ns/activitystreams\", \"type\": \"Announce\", \"to\": [\"https://relay.101010.pl/followers\"], \"actor\": \"https://relay.101010.pl/actor\", \"object\": \"https://incitter.online/users/Sezaisrailoglu/statuses/105601248545971833\", \"id\": \"https://relay.101010.pl/activities/b84a041e-a9be-4fac-b114-26ed2fca6074\"}",null],"jid":"fefe6b40a6789792e9815a18","created_at":1611361613.5014067,"enqueued_at":1611361613.501452},"jobstr":"{\"retry\":8,\"queue\":\"default\",\"backtrace\":true,\"class\":\"ActivityPub::ProcessingWorker\",\"args\":[4235,\"{\\\"@context\\\": \\\"https://www.w3.org/ns/activitystreams\\\", \\\"type\\\": \\\"Announce\\\", \\\"to\\\": [\\\"https://relay.101010.pl/followers\\\"], \\\"actor\\\": \\\"https://relay.101010.pl/actor\\\", \\\"object\\\": \\\"https://incitter.online/users/Sezaisrailoglu/statuses/105601248545971833\\\", \\\"id\\\": \\\"https://relay.101010.pl/activities/b84a041e-a9be-4fac-b114-26ed2fca6074\\\"}\",null],\"jid\":\"fefe6b40a6789792e9815a18\",\"created_at\":1611361613.5014067,\"enqueued_at\":1611361613.501452}"}
sidekiq_1    | 2021-01-23T04:23:16.222Z pid=7 tid=q0l7 WARN: Mastodon::RaceConditionError: Mastodon::RaceConditionError
sidekiq_1    | 2021-01-23T04:23:16.223Z pid=7 tid=q0l7 WARN: /opt/mastodon/app/lib/activitypub/activity/announce.rb:28:in `block in perform'
sidekiq_1    | /opt/mastodon/vendor/bundle/ruby/2.7.0/gems/mario-redis-lock-1.2.1/lib/redis_lock.rb:49:in `acquire'
sidekiq_1    | /opt/mastodon/app/lib/activitypub/activity/announce.rb:7:in `perform'
sidekiq_1    | 2021-01-23T04:23:16.223Z pid=7 tid=q2fj class=ActivityPub::ProcessingWorker jid=a0d43339eb669416dfb3a358 INFO: start
sidekiq_1    | 2021-01-23T04:23:16.383Z pid=7 tid=q1tb class=ActivityPub::ProcessingWorker jid=cef5beec434cd328ba98a136 elapsed=10.048 INFO: fail
sidekiq_1    | 2021-01-23T04:23:16.383Z pid=7 tid=q1tb WARN: {"context":"Job raised exception","job":{"retry":8,"queue":"default","backtrace":true,"class":"ActivityPub::ProcessingWorker","args":[4235,"{\"@context\": \"https://www.w3.org/ns/activitystreams\", \"type\": \"Announce\", \"to\": [\"https://relay.101010.pl/followers\"], \"actor\": \"https://relay.101010.pl/actor\", \"object\": \"https://mstdn.foxfam.club/users/TheSun/statuses/105600889136661089\", \"id\": \"https://relay.101010.pl/activities/e3cc1d72-5125-4f0a-b182-5ac2cf4999d0\"}",null],"jid":"cef5beec434cd328ba98a136","created_at":1611361619.8029497,"enqueued_at":1611361619.8029943},"jobstr":"{\"retry\":8,\"queue\":\"default\",\"backtrace\":true,\"class\":\"ActivityPub::ProcessingWorker\",\"args\":[4235,\"{\\\"@context\\\": \\\"https://www.w3.org/ns/activitystreams\\\", \\\"type\\\": \\\"Announce\\\", \\\"to\\\": [\\\"https://relay.101010.pl/followers\\\"], \\\"actor\\\": \\\"https://relay.101010.pl/actor\\\", \\\"object\\\": \\\"https://mstdn.foxfam.club/users/TheSun/statuses/105600889136661089\\\", \\\"id\\\": \\\"https://relay.101010.pl/activities/e3cc1d72-5125-4f0a-b182-5ac2cf4999d0\\\"}\",null],\"jid\":\"cef5beec434cd328ba98a136\",\"created_at\":1611361619.8029497,\"enqueued_at\":1611361619.8029943}"}
```

### Steps to reproduce the problem

1. Log in to instance.
2. Drag an image file (51.8kb JPG) into the compose box.
3. Observe that upload never completes.

### Specifications

I'm running docker tag `tootsuite/mastodon:v3.3.0`.

I replicated in Firefox 84.0.2 (64-bit), both in a normal window, and a private window with add-ons disabled, and also in Version 87.0.4280.66 (Official Build) Built on Ubuntu , running on Ubuntu 18.04 (64-bit).

**Comment by ericlathrop** [comment]
I've tried restarting all the containers, which didn't fix it. I suspect there's a problem inside of sidekiq somewhere, but I don't know how to debug that further.

**Comment by ThibG** [comment]
The `Mastodon::RaceConditionError` thing sounds fishy but also doesn't seem to be directly related. Could you check:
1. if there are stuck sidekiq jobs (you can check that from `https://yourinstance.tld/sidekiq/busy`)
2. if there's anything fishy in redis' logs (`journalctl -u redis -n 50`)

**Comment by ericlathrop** [comment]
There seems to always be 5 jobs running, but they're never older than 30 seconds. There's 86,608 jobs enqueued, which seems excessive. The majority seems to be `LinkCrawlWorker`.

![Screenshot from 2021-01-23 15-12-41](https://user-images.githubusercontent.com/530891/105613083-39b78600-5d8e-11eb-9c38-9ea9c03d4ecf.png)

None of the containers seem to have `journalctl` installed. The `docker-compose logs redis` look like:

```
redis_1      | 1:M 23 Jan 2021 20:02:58.048 * 100 changes in 300 seconds. Saving...
redis_1      | 1:M 23 Jan 2021 20:02:58.049 * Background saving started by pid 20343
redis_1      | 20343:C 23 Jan 2021 20:02:58.306 * DB saved on disk
redis_1      | 20343:C 23 Jan 2021 20:02:58.308 * RDB: 1 MB of memory used by copy-on-write
redis_1      | 1:M 23 Jan 2021 20:02:58.350 * Background saving terminated with success
```

**Comment by ericlathrop** [comment]
It seems like the number of enqueued jobs keeps increasing...

Btw, I'm running a small single-user instance.

**Comment by ericlathrop** [comment]
I spoke with some people on Freenode, and discovered a bunch of `ActivityPub::ProcessingWorker` messages in the `push` queue related to 2 relays in particular. These messages were taking ~15 seconds each, so I disabled those 2 relays, and my "Enqueued" job count slowly started decreasing. It was then suggested that I needed more sidekiq workers, so I bumped it from the default 5 to 15, and it started processing the messages more quickly. I went down to a high count of ~88,000 jobs to now ~61,000 jobs, and now images are uploading correctly.
